android/src/main/java/com/mapbox/mapboxgl/MapboxMapController.java

```diff
@@ -958,6 +967,69 @@ public void onFailure(@NonNull Exception exception) {
         result.success(null);
         break;
       }
+      case "style#getLayers": {
+        if (style == null) {
+          result.error("STYLE IS NULL", "The style is null. Has onStyleLoaded() already been invoked?", null);
+        }
+
+        List<Layer> layers = style.getLayers();
+        List<String> layerIds = new ArrayList<String>(layers.size());
+        for (Layer layer : layers) {
+          layerIds.add(layer.getId());
+        }
+
+        result.success(layerIds);
+        break;
+      }
+      case "style#showLayer": {
+        if (style == null) {
+          result.error("STYLE IS NULL", "The style is null. Has onStyleLoaded() already been invoked?", null);
+        }
+
+        String layerId = call.argument("imageLayerId");
+        Layer layer = style.getLayer(layerId);
+
+        if (layer == null)
+          return;
+
+        layer.setProperties(visibility(VISIBLE));
+        result.success(null);
+        break;
+      }
+      case "style#hideLayer": {
+        if (style == null) {
+          result.error("STYLE IS NULL", "The style is null. Has onStyleLoaded() already been invoked?", null);
+        }
+
+        String layerId = call.argument("imageLayerId");
+        Layer layer = style.getLayer(layerId);
+
+        if (layer == null)
+          return;
```

Comment:
break? having a large function like this return from the middle is weird? Might also be a good idea to set result.error so that this doesn’t  fail silently